### PR TITLE
Append val to solr array instead of replacing when appropriate.

### PR DESCRIPTION
**Pull request by hackmastera**
Fixes projecthydra/active_fedora#870.

### CODE REVIEW

File: lib/active_fedora/loadable_from_json.rb [hunk 1]
```diff
@@ -45,7 +45,11 @@ def to_s
       # It is expected that the singular filter gets applied after fetching the value from this
       # resource, so cast everything back to an array.
       def set_value(k, v)
-        @hash[k] = Array(v)
+        if (!@hash[k].nil? && @hash[k].is_a?(Array))
```
Comment (jcoyne) [inline lib/active_fedora/loadable_from_json.rb]: No need to trap for nil here:
```
nil.is_a? Array
=> false
```

Comment (jcoyne) [inline lib/active_fedora/loadable_from_json.rb]: I'm not sure this refactor changes anything.  Where is the difference?

Comment (jcoyne) [inline lib/active_fedora/loadable_from_json.rb]: Oh, I see it's appending vs replacing. Why would we append in a method called "set_value"?

**Comment by hackmastera** [comment]
@jcoyne thank you for your comments. How does this look?

**Comment by mjgiarlo** [comment]
Looks clean to me but this is @jcoyne's wheelhouse.

File: spec/integration/solr_instance_loader_spec.rb [hunk 2]
```diff
@@ -43,6 +57,14 @@ class Foo < ActiveFedora::Base
       it "should not be mutable" do
         expect { subject.title = 'Foo' }.to raise_error ActiveFedora::ReadOnlyRecord
       end
+
+      it "loads the correct number of nested attributes", focus: true do
```
Comment (jcoyne) [inline spec/integration/solr_instance_loader_spec.rb]: why is `focus: true` set here?

File: spec/integration/solr_instance_loader_spec.rb [hunk 3]
```diff
@@ -43,6 +57,14 @@ class Foo < ActiveFedora::Base
...
+        expect(subject.date_ids.count).to eq 2
+      end
+
+      it "loads the correct number of regular attributes", focus: true do
```
Comment (jcoyne) [inline spec/integration/solr_instance_loader_spec.rb]: why `focus`?

Comment (hackmastera) [inline spec/integration/solr_instance_loader_spec.rb]: Ugh, good catch! ha, AF's rspec wasn't even set up for this so I didn't notice it was there.

**Pull request merged**